pkg/api/mock/mock_api_iterator.go

```diff
@@ -0,0 +1,74 @@
+package mock
+
+import (
+	"errors"
+
+	"go.anx.io/go-anxcloud/pkg/api/types"
+)
+
+var (
+	// ErrPageSizeCannotBeZero is returned when newMockPageIter is getting called with itemsPerPage = 0 to prevent devision by zero panic
+	ErrPageSizeCannotBeZero = errors.New("mockPageIter itemsPerPage cannot be zero")
```

Comment:
hmm... what does the real Engine API do in such a case? :D

Reply:
Most endpoints seem to have no limit when page size (limit) is set to zero. But ENGSUP-6081 say's we shouldn't use it. So I think it's okay to return an error.